packages/twenty-server/src/workspace/workspace-query-builder/factories/record-position-query.factory.ts

```diff
@@ -1,21 +1,74 @@
 import { Injectable } from '@nestjs/common';
 
-import { ObjectMetadataInterface } from 'src/metadata/field-metadata/interfaces/object-metadata.interface';
+export enum RecordPositionQueryType {
+  GET = 'GET',
+  UPDATE = 'UPDATE',
+}
 
 @Injectable()
 export class RecordPositionQueryFactory {
   async create(
+    recordPositionQueryType: RecordPositionQueryType,
+    positionValue: 'first' | 'last' | number,
+    objectMetadata: { isCustom: boolean; nameSingular: string },
+    dataSourceSchema: string,
+    recordId?: string,
+  ): Promise<string> {
+    const name =
+      (objectMetadata.isCustom ? '_' : '') + objectMetadata.nameSingular;
```

Comment:
nit: We should probably have a helper for that for when we will remove this logic